pkg/scan/process_image.go

```diff
@@ -271,21 +271,27 @@ func ScanFile(s *Scanner, f *os.File, iocs *[]output.IOCFound, layer string) err
 		}
 		err = yrScanner.ScanMem(buf)
 		if err != nil {
-			fmt.Println("Scan File Mmory Error", err)
+			logrus.Errorf("yara: %s: Error scanning file, error=%s", fileName, err.Error())
 			return filepath.SkipDir
 		}
 
 	}
 	var iocsFound []output.IOCFound
 	totalMatchesStringData := make([]string, 0)
 	for _, m := range matches {
-		matchesStringData := make([]string, len(m.Strings))
+		matchesStringData := make(map[string]struct{})
 		for _, str := range m.Strings {
-			if !strings.Contains(strings.Join(matchesStringData, " "), string(str.Data)) {
-				matchesStringData = append(matchesStringData, string(str.Data))
+			if _, exists := matchesStringData[string(str.Data)]; !exists {
+				matchesStringData[string(str.Data)] = struct{}{}
 				totalMatchesStringData = append(totalMatchesStringData, string(str.Data))
 			}
 		}
```

Comment:
This looks completely unnecessary - to be checked but if `m.Strings` are already unique, we can skip this entirely. If we don't we simply need to run `Unique` on the slice and we can avoid the `map` entirely.

The only thing we need to do is transform the `[]byte` into `string` , which we can do cheaply via:
```
func StringToBytes(s string) []byte {
	return unsafe.Slice(unsafe.StringData(s), len(s))
}
```

This will run into trouble if `m.Strings` is modified after the conversion (to be checked). If the `[]byte` are dropped, then we can avoid all the `string` allocations.

Reply:
I agree. IMO `m.Strings` are not unique. We need to make it unique (slice set), and map was quickest way I could think of. I am not sure about any other way.

```
func StringToBytes(s string) []byte {
	return unsafe.Slice(unsafe.StringData(s), len(s))
}
```
This we can use to transform.

Reply:
updated the code. please check.